backend/src/main/java/gov/cdc/usds/simplereport/db/repository/ApiUserRepository.java

```diff
@@ -30,4 +33,37 @@ public interface ApiUserRepository extends EternalSystemManagedEntityRepository<
 
   @Query(BASE_QUERY + " and loginEmail IN :emails" + NAME_ORDER)
   List<ApiUser> findAllByLoginEmailInOrderByName(Collection<String> emails);
+
+  String API_USER_ROLE_LEFT_JOIN = " LEFT JOIN api_user_role aur ON aur.apiUser = e";
+  String BY_ORG_AND_UNDELETED_USER = " WHERE aur.organization = :org AND e.isDeleted = false";
+  String JOINED_NAME_ORDER =
+      " order by e.nameInfo.lastName, e.nameInfo.firstName, e.nameInfo.middleName, e.internalId";
+
+  @Query(
+      value =
+          "from #{#entityName} e"
+              + API_USER_ROLE_LEFT_JOIN
+              + BY_ORG_AND_UNDELETED_USER
+              + JOINED_NAME_ORDER)
+  List<ApiUser> findAllByOrganization(Organization org);
+
+  @Query(
+      value =
+          "from #{#entityName} e"
+              + API_USER_ROLE_LEFT_JOIN
+              + BY_ORG_AND_UNDELETED_USER
+              + " AND aur.role = :role"
+              + JOINED_NAME_ORDER)
+  List<ApiUser> findAllByOrganizationAndRole(Organization org, OrganizationRole role);
+
+  @Query(
+      value =
+          "from #{#entityName} e"
+              + API_USER_ROLE_LEFT_JOIN
+              + " LEFT JOIN api_user_facility auf ON auf.apiUser = e"
+              + " WHERE auf.facility = :facility"
+              + " AND e.isDeleted = false"
+              + " AND (aur.role = 'USER' OR aur.role = 'ENTRY_ONLY')"
+              + " GROUP BY e")
+  List<ApiUser> findAllBySingleFacilityAccess(Facility facility);
```

Comment:
I think the way you use this query to get all the users and then check for facility count is great but my nit is to change this method name since this technically doesn't do the "single facility" filtering yet

or make this method something like `findAllByFacilityAndRoles(Facility facility, Collection<OrganizationRole> roles)` and pass in the specific roles

Reply:
Good call! I'll change that!

Reply:
Done!